optimum/neuron/distributed/parallel_layers.py

```diff
@@ -250,6 +275,41 @@ def transform(
         else:
             layer_qualified_name = ""
 
+        if cls.NUM_ATTENTION_HEADS_NAME is None:
+            num_attention_heads_name = normalized_config.NUM_ATTENTION_HEADS
+        else:
+            num_attention_heads_name = cls.NUM_ATTENTION_HEADS_NAME
+
+        if not hasattr(layer, num_attention_heads_name):
+            raise AttributeError(f"The {type(layer)} layer has not attribute {num_attention_heads_name}.")
+
+        num_attention_heads = getattr(layer, num_attention_heads_name)
+
+        if cls.ALL_HEAD_SIZE_NAME is None:
+            all_head_size_name = normalized_config.ALL_HEAD_SIZE_NAME
+        else:
+            all_head_size_name = cls.ALL_HEAD_SIZE_NAME
+
+        if not hasattr(layer, all_head_size_name):
+            raise AttributeError(f"The {type(layer)} layer has not attribute {all_head_size_name}.")
+
+        if cls.NUM_KEY_VALUE_HEADS_NAME is not None:
+            num_key_value_heads = getattr(layer, cls.NUM_KEY_VALUE_HEADS_NAME)
+            if num_key_value_heads % tp_size != 0 and tp_size % num_key_value_heads != 0:
+                raise NotImplementedError(
```

Comment:
Are you going to support the non-divisible case in the future? If not probably not use `NotImplementedError`.